### PR TITLE
WIP added pages to detect and correct Issues / Change Data

### PR DESCRIPTION
**Pull request by MaxTrautwein**
This is not ready for merge.
It needs more work, but the priority's have changed. until then this is on hold

### CODE REVIEW

File: backend/scripts/Interfaces.py [hunk 1]
```diff
@@ -13,7 +13,42 @@ def __init__(self, item_id: int, name: str, price: float, manufacturer: str, col
...
+    def __init__(self, json):
+        self.id = json["id"]
+        self.name = json["name"]
+        self.price = json["price"]
+        self.manufacturer = json["manufacturer"]
+        self.color = json["color"]
+        self.details = json["details"]
+        self.size = json["size"]
+        self.tax = json["tax"]
+        self.count = json["count"] # NOT Part of the DB but used for Transactions
```
Comment (Imd0odO) [inline backend/scripts/Interfaces.py:30]: why is self.count set to current stock instead to the value in cart?

File: backend/scripts/db.py [hunk 2]
```diff
@@ -88,6 +76,23 @@ def AddNewItem(Data):
...
+        IDs.append(item[0])
+    execute("select  id from items where cnt < items.min_cnt;")
+    data = cur.fetchall()
+    for item in data:
+        IDs.append(item[0])
+    return IDs
+
+def UpdateItem(ItemData: Interfaces.Item):
+    execute("update items set (name, cnt, price, manufacturer, color, min_cnt, details, size, bon_name) = "+
+            "('{}',{},{},'{}','{}',{},'{}','{}','{}') where id = {}"
```
Comment (Imd0odO) [inline backend/scripts/db.py:93]: use f"{variable}, {variable}" for clarification. (More easily to understand that way)

File: backend/scripts/db.py [hunk 3]
```diff
@@ -88,6 +76,23 @@ def AddNewItem(Data):
...
+            .format(ItemData.name,ItemData.cnt,ItemData.price,ItemData.manufacturer,ItemData.color,ItemData.min_cnt,ItemData.details,ItemData.size,ItemData.bon_name,ItemData.id))
+
 class NetworkError(Exception):
```
Comment (Imd0odO) [inline backend/scripts/db.py:96]: Exceptions to top of the file

File: backend/scripts/main.py [hunk 4]
```diff
@@ -77,14 +78,19 @@ def GetItem():
...
+
 @app.route('/make_sale', methods=['POST'])
 def MakeSale():
     content = request.json
     Items = []
     #Convert Basket from Frontend into compatible format for Printing
     for item in content:
         newItem = db.GetItemByID(item["id"])
-        newItem.SetCount(item['cnt'])
+        newItem.count = item['cnt']
```
Comment (Imd0odO) [inline backend/scripts/main.py:93]: why are cnt and count refering to the same value (cnt == stock;   count == basket)?
Edit count where cnt is set at the moment.

**Comment by MaxTrautwein** [comment]
Closing as with the New Layout this stale PR is of no use

**Pull request closed without merging**